src/map/packet_system.cpp

```diff
@@ -6654,6 +6668,8 @@ void SmallPacket0x0E7(map_session_data_t* const PSession, CCharEntity* const PCh
         return;
     }
 
+    charutils::SaveCharEquip(PChar);
```

Comment:
If 2x swaps happen in quick succession and then a player logs out, the most recent gear won't be saved by this?

You could add a forceSave bool to the call to make sure it saves

Reply:
I'm not sure why it would be ignored here? The rate-limit is only being imposed by incoming 050 and 051, `SaveCharEquip()` isn't rate limited at all inside of the actual function

Reply:
Ah I see what you mean, please disregard